### PR TITLE
python310Packages.nbconvert: use mistune 2.x

### LINKED ISSUE #186871: `python310Packages.nbconvert` build failure on x86_64-linux as of `a093c35b`
## Issue description
Build of `python310Packages.nbconvert` failed on x86_64-linux as of a093c35beee0a64ad68312d747a4c49eaa5b1105. This is currently breaking `python3Packages.einops`.

```
error: builder for '/nix/store/4277frbm7rmf2pw9nk3fxnv4zbn5w40w-python3.10-nbconvert-6.5.0.drv' failed with exit code 1;
       last 10 log lines:
       > Requirement already satisfied: jupyter-core>=4.7 in /nix/store/ap0g73di2873pj7m28i6ww8zn0cnvihg-python3.10-jupyter_core-4.9.2/lib/python3.10/site-packages (from nbconvert==6.5.0) (4.9.2)
       > Requirement already satisfied: bleach in /nix/store/1agcqr28acfw6cf1cag4da7f9rx685dp-python3.10-bleach-5.0.1/lib/python3.10/site-packages (from nbconvert==6.5.0) (5.0.1)
       > Requirement already satisfied: nbformat>=5.1 in /nix/store/h2k2q522y8c48szz39wgipc673pf62vf-python3.10-nbformat-5.4.0/lib/python3.10/site-packages (from nbconvert==6.5.0) (5.4.0)
       > Requirement already satisfied: pygments>=2.4.1 in /nix/store/czicm46iryrk0rssz4zs8rn788x4ni42-python3.10-pygments-2.12.0/lib/python3.10/site-packages (from nbconvert==6.5.0) (2.12.0)
       > Requirement already satisfied: entrypoints>=0.2.2 in /nix/store/6r1vdaz2vj5gzlnb9spz8inv0r7gc3iq-python3.10-entrypoints-0.4/lib/python3.10/site-packages (from nbconvert==6.5.0) (0.4)
       > Requirement already satisfied: packaging in /nix/store/rcvz5ywnm8fla013hqsg01imz2i41k7p-python3.10-packaging-21.3/lib/python3.10/site-packages (from nbconvert==6.5.0) (21.3)
       > Requirement already satisfied: jinja2>=3.0 in /nix/store/nd6636vpb09q2f9p79a5w3kdhv1qbw2j-python3.10-Jinja2-3.1.2/lib/python3.10/site-packages (from nbconvert==6.5.0) (3.1.2)
       > ERROR: Could not find a version that satisfies the requirement mistune<2,>=0.8.1 (from nbconvert) (from versions: none)
       > ERROR: No matching distribution found for mistune<2,>=0.8.1
       > 
       For full logs, run 'nix log /nix/store/4277frbm7rmf2pw9nk3fxnv4zbn5w40w-python3.10-nbconvert-6.5.0.drv'.
error: 1 dependencies of derivation '/nix/store/jpb7bh0g1j156vvziclksfxmrzs9fc25-python3.10-einops-0.4.1.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2864366564)

python3Packages.einops maintainers cc: @yl3dy
python310Packages.nbconvert maintainers cc: @FRidh
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit a093c35beee0a64ad68312d747a4c49eaa5b1105
2. Run `nix-build -A python310Packages.nbconvert`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-1014-azure, Ubuntu, 20.04.4 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - nixpkgs: `/nix/store/xbmc7mwndkbykkcdkgwvl2i8dx43db6x-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2864366564) that created this issue.
- Internal tags: -5805513848243597698 3278788159294963055

### PR DESCRIPTION
**Pull request by smancill**
###### Description of changes

mistune 0.8 was removed by #186272.

Patch the sources to use mistune 2.x.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by smancill** [comment]
@ofborg build python310Packages.jupyter

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 186804` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>45 packages failed to build:</summary>
  <ul>
    <li>manim</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.folium</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.nbclassic</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>spyder (python310Packages.spyder)</li>
    <li>python310Packages.treex</li>
    <li>python39Packages.arviz</li>
    <li>python39Packages.dalle-mini</li>
    <li>python39Packages.elegy</li>
    <li>python39Packages.folium</li>
    <li>python39Packages.jupyterlab</li>
    <li>python39Packages.jupyterlab-git</li>
    <li>python39Packages.jupyterlab-lsp</li>
    <li>python39Packages.jupyterlab_server</li>
    <li>python39Packages.jupytext</li>
    <li>python39Packages.modeled</li>
    <li>python39Packages.nbclassic</li>
    <li>python39Packages.osmnx</li>
    <li>python39Packages.pymc</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aer</li>
    <li>python39Packages.qiskit-finance</li>
    <li>python39Packages.qiskit-ibmq-provider</li>
    <li>python39Packages.qiskit-ignis</li>
    <li>python39Packages.qiskit-nature</li>
    <li>python39Packages.qiskit-optimization</li>
    <li>python39Packages.robotframework-tools</li>
    <li>python39Packages.spyder</li>
    <li>python39Packages.treex</li>
    <li>wolfram-notebook</li>
  </ul>
</details>
<details>
  <summary>76 packages built:</summary>
  <ul>
    <li>ihaskell</li>
    <li>jupyter</li>
    <li>opsdroid</li>
    <li>python310Packages.Nikola</li>
    <li>python310Packages.Pweave</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.datashader</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter-lsp</li>
    <li>python310Packages.jupyter-server-mathjax</li>
    <li>python310Packages.jupyter-sphinx</li>
    <li>python310Packages.jupyter_server</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterlab_launcher</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.livelossplot</li>
    <li>python310Packages.nbconflux</li>
    <li>python310Packages.nbconvert</li>
    <li>python310Packages.nbdime</li>
    <li>python310Packages.nbsmoke</li>
    <li>python310Packages.nbsphinx</li>
    <li>python310Packages.nix-kernel</li>
    <li>python310Packages.notebook</li>
    <li>python310Packages.notedown</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.papermill</li>
    <li>python310Packages.phik</li>
    <li>python310Packages.qiskit-machine-learning</li>
    <li>python310Packages.qiskit-terra</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python39Packages.Nikola</li>
    <li>python39Packages.Pweave</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.augmax</li>
    <li>python39Packages.batchspawner</li>
    <li>python39Packages.datashader</li>
    <li>python39Packages.dockerspawner</li>
    <li>python39Packages.einops</li>
    <li>python39Packages.jupyter</li>
    <li>python39Packages.jupyter-lsp</li>
    <li>python39Packages.jupyter-server-mathjax</li>
    <li>python39Packages.jupyter-sphinx</li>
    <li>python39Packages.jupyter_server</li>
    <li>python39Packages.jupyterhub</li>
    <li>python39Packages.jupyterhub-ldapauthenticator</li>
    <li>python39Packages.jupyterhub-systemdspawner</li>
    <li>python39Packages.jupyterhub-tmpauthenticator</li>
    <li>python39Packages.jupyterlab_launcher</li>
    <li>python39Packages.livelossplot</li>
    <li>python39Packages.moretools</li>
    <li>python39Packages.nbconflux</li>
    <li>python39Packages.nbconvert</li>
    <li>python39Packages.nbdime</li>
    <li>python39Packages.nbsmoke</li>
    <li>python39Packages.nbsphinx</li>
    <li>python39Packages.nix-kernel</li>
    <li>python39Packages.notebook</li>
    <li>python39Packages.notedown</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.papermill</li>
    <li>python39Packages.phik</li>
    <li>python39Packages.qiskit-machine-learning</li>
    <li>python39Packages.qiskit-terra</li>
    <li>python39Packages.vega</li>
    <li>python39Packages.xarray-einstats</li>
    <li>python39Packages.zetup</li>
    <li>sage</li>
    <li>sageWithDoc</li>
  </ul>
</details>

**Comment by smancill** [comment]
Too many failing packages still, that's not good.

**Comment by smancill** [comment]
@ofborg build python310Packages.jupyterlab python310Packages.folium

**Comment by smancill** [comment]
A quick glance shows that, along with `nbclassic`, the failing packages were broken by the previous Python updates:
 
- https://hydra.nixos.org/job/nixpkgs/trunk/python310Packages.arviz.x86_64-linux
- https://hydra.nixos.org/job/nixpkgs/trunk/python310Packages.folium.x86_64-linux
- https://hydra.nixos.org/job/nixpkgs/trunk/python310Packages.pymc.x86_64-linux

**Comment by anthonyroussel** [comment]
Result of `nixpkgs-review pr 186804` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>51 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>manim: log was empty</li>
<li>python310Packages.arviz: log was empty</li>
<li>python310Packages.batchspawner: log was empty</li>
<li>python310Packages.dalle-mini: log was empty</li>
<li>python310Packages.dockerspawner: log was empty</li>
<li>python310Packages.elegy: log was empty</li>
<li>python310Packages.folium: log was empty</li>
<li>python310Packages.jupyterhub: log was empty</li>
<li>python310Packages.jupyterhub-ldapauthenticator: log was empty</li>
<li>python310Packages.jupyterhub-systemdspawner: log was empty</li>
<li>python310Packages.jupyterhub-tmpauthenticator: log was empty</li>
<li>python310Packages.jupyterlab: log was empty</li>
<li>python310Packages.jupyterlab-git: log was empty</li>
<li>python310Packages.jupyterlab-lsp: log was empty</li>
<li>python310Packages.jupytext: log was empty</li>
<li>python310Packages.nbclassic: log was empty</li>
<li>python310Packages.oauthenticator: log was empty</li>
<li>python310Packages.osmnx: log was empty</li>
<li>python310Packages.pymc: log was empty</li>
<li>python310Packages.qiskit: log was empty</li>
<li>python310Packages.qiskit-aer: log was empty</li>
<li>python310Packages.qiskit-finance: log was empty</li>
<li>python310Packages.qiskit-ibmq-provider: log was empty</li>
<li>python310Packages.qiskit-ignis: log was empty</li>
<li>python310Packages.qiskit-nature: log was empty</li>
<li>python310Packages.qiskit-optimization: log was empty</li>
<li>spyder: log was empty</li>
<li>python310Packages.treex: log was empty</li>
<li>python39Packages.arviz: log was empty</li>
<li>python39Packages.dalle-mini: log was empty</li>
<li>python39Packages.elegy: log was empty</li>
<li>python39Packages.folium: log was empty</li>
<li>python39Packages.jupyterlab: log was empty</li>
<li>python39Packages.jupyterlab-git: log was empty</li>
<li>python39Packages.jupyterlab-lsp: log was empty</li>
<li>python39Packages.jupyterlab_server: log was empty</li>
<li>python39Packages.jupytext: log was empty</li>
<li>python39Packages.modeled: log was empty</li>
<li>python39Packages.nbclassic: log was empty</li>
<li>python39Packages.osmnx: log was empty</li>
<li>python39Packages.pymc: log was empty</li>
<li>python39Packages.qiskit: log was empty</li>
<li>python39Packages.qiskit-aer: log was empty</li>
<li>python39Packages.qiskit-finance: log was empty</li>
<li>python39Packages.qiskit-ibmq-provider: log was empty</li>
<li>python39Packages.qiskit-ignis: log was empty</li>
<li>python39Packages.qiskit-nature: log was empty</li>
<li>python39Packages.qiskit-optimization: log was empty</li>
<li>python39Packages.robotframework-tools: log was empty</li>
<li>python39Packages.spyder: log was empty</li>
<li>python39Packages.treex: log was empty</li>
  </ul>
</details>
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>(python310Packages.spyder): log was empty</li>
<li>wolfram-notebook: log was empty</li>
  </ul>
</details>
<details>
<summary>
67 packages built:
</summary>
<ul>
<li>ihaskell</li>
<li>jupyter</li>
<li>opsdroid</li>
<li>python310Packages.Nikola</li>
<li>python310Packages.Pweave</li>
<li>python310Packages.ansible-kernel</li>
<li>python310Packages.augmax</li>
<li>python310Packages.datashader</li>
<li>python310Packages.einops</li>
<li>python310Packages.jupyter</li>
<li>python310Packages.jupyter-lsp</li>
<li>python310Packages.jupyter-server-mathjax</li>
<li>python310Packages.jupyter-sphinx</li>
<li>python310Packages.jupyter_server</li>
<li>python310Packages.jupyterlab_launcher</li>
<li>python310Packages.jupyterlab_server</li>
<li>python310Packages.livelossplot</li>
<li>python310Packages.nbconflux</li>
<li>python310Packages.nbconvert</li>
<li>python310Packages.nbdime</li>
<li>python310Packages.nbsmoke</li>
<li>python310Packages.nbsphinx</li>
<li>python310Packages.nix-kernel</li>
<li>python310Packages.notebook</li>
<li>python310Packages.notedown</li>
<li>python310Packages.papermill</li>
<li>python310Packages.phik</li>
<li>python310Packages.qiskit-machine-learning</li>
<li>python310Packages.qiskit-terra</li>
<li>python310Packages.vega</li>
<li>python310Packages.xarray-einstats</li>
<li>python39Packages.Nikola</li>
<li>python39Packages.Pweave</li>
<li>python39Packages.ansible-kernel</li>
<li>python39Packages.augmax</li>
<li>python39Packages.batchspawner</li>
<li>python39Packages.datashader</li>
<li>python39Packages.dockerspawner</li>
<li>python39Packages.einops</li>
<li>python39Packages.jupyter</li>
<li>python39Packages.jupyter-lsp</li>
<li>python39Packages.jupyter-server-mathjax</li>
<li>python39Packages.jupyter-sphinx</li>
<li>python39Packages.jupyter_server</li>
<li>python39Packages.jupyterhub</li>
<li>python39Packages.jupyterhub-ldapauthenticator</li>
<li>python39Packages.jupyterhub-systemdspawner</li>
<li>python39Packages.jupyterhub-tmpauthenticator</li>
<li>python39Packages.jupyterlab_launcher</li>
<li>python39Packages.livelossplot</li>
<li>python39Packages.moretools</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbconvert</li>
<li>python39Packages.nbdime</li>
<li>python39Packages.nbsmoke</li>
<li>python39Packages.nbsphinx</li>
<li>python39Packages.nix-kernel</li>
<li>python39Packages.notebook</li>
<li>python39Packages.notedown</li>
<li>python39Packages.oauthenticator</li>
<li>python39Packages.papermill</li>
<li>python39Packages.phik</li>
<li>python39Packages.qiskit-machine-learning</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.vega</li>
<li>python39Packages.xarray-einstats</li>
<li>python39Packages.zetup</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ihaskell:
<details>

Package is missing a meta section.
If the package is using runCommand please make sure to inherit or create a meta section.

got build log for '/nix/store/4ss4y20jchfj2v2808m19kgryawn6v49-ihaskell-with-packages' from 'daemon'
When evaluating attribute ‘ihaskell’:
warning: unused-argument
Unused argument: `paths`.
Near pkgs/build-support/trivial-builders.nix:433:12:

```
    |
433 |          , paths
    |            ^
```
</details>
</details>

File: pkgs/development/python-modules/nbconvert/default.nix [hunk 1]
```diff
@@ -30,10 +31,21 @@ buildPythonPackage rec {
   # various exporter templates
   patches = [
     ./templates.patch
+
+    # Use mistune 2.x
+    (fetchpatch {
+      url = "https://github.com/jupyter/nbconvert/commit/e870d9a4a61432a65bee5466c5fa80c9ee28966e.patch";
```
Comment (risicle) [inline pkgs/development/python-modules/nbconvert/default.nix:37]:
```suggestion
      name = "mistune-2.x-support.patch";
      url = "https://github.com/jupyter/nbconvert/commit/e870d9a4a61432a65bee5466c5fa80c9ee28966e.patch";
```

**Comment by risicle** [comment]
As others are finding, what with the muddle of recent breakages it's hard to tell, but I don't *think* this causes any new breakages for `nixpkgs-review` on macos 10.15.

**Review by dotlambda — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Some tests related to folium fail.

**Comment by onny** [comment]
There's also a new version of nbconvert available: 6.5.3

**Comment by smancill** [comment]
> Some tests related to folium fail.

It was already broken, so it's unrelated to this PR: https://hydra.nixos.org/job/nixpkgs/trunk/python310Packages.folium.x86_64-linux

**Comment by smancill** [comment]
> There's also a new version of nbconvert available: 6.5.3

Support for mistune 2.x was added in the 7.x RC, so that new 6.5.3 version will not affect this.

File: pkgs/development/python-modules/nbconvert/default.nix (hunk 1 above)
Comment (smancill) [reply pkgs/development/python-modules/nbconvert/default.nix:38]: Done.

**Pull request merged**